cmd/agentctl2/utils/common_utils.go

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2017 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"os"
+	"strings"
+
+	"github.com/gogo/protobuf/proto"
+
+	"fmt"
+
+	"github.com/ligato/cn-infra/db/keyval"
+	"github.com/ligato/cn-infra/db/keyval/etcd"
+	"github.com/ligato/cn-infra/db/keyval/kvproto"
+	"github.com/ligato/cn-infra/logging"
+	"github.com/ligato/cn-infra/logging/logrus"
+	"github.com/ligato/cn-infra/servicelabel"
+)
+
+const (
+	ACLPath          = "config/vpp/acls/v2/acl/"
+	InterfacePath    = "config/vpp/v2/interfaces/"
+	BridgeDomainPath = "config/vpp/l2/v2/bridge-domain/"
+	FibTablePath     = "config/vpp/l2/v2/fib/"
+	XConnectPath     = "config/vpp/l2/v2/xconnect/"
+	ARPPath          = "config/vpp/v2/arp/"
+	RoutePath        = "config/vpp/v2/route/"
+	ProxyARPPath     = "config/vpp/v2/proxyarp-global"
+	IPScanneightPath = "config/vpp/v2/ipscanneigh-global"
+	NATPath          = "config/vpp/nat/v2/nat44-global"
+	DNATPath         = "config/vpp/nat/v2/dnat44/"
+	IPSecPolicyPath  = "config/vpp/ipsec/v2/spd/"
+	IPSecAssociate   = "config/vpp/ipsec/v2/sa/"
+)
+
+const (
+	lInterfacePath = "config/linux/interfaces/v2/interface/"
+	lARPPath       = "config/linux/l4/v2/arp/"
+	lRoutePath     = "config/linux/l3/v2/route/"
+)
+
+// Common exit flags
+const (
+	ExitSuccess = iota
+	ExitError
+	ExitBadConnection
+	ExitInvalidInput
+	ExitBadFeature
+	ExitInterrupted
+	ExitIO
+	ExitBadArgs = 128
+)
+
+// ParseKey parses the etcd Key for the microservice label and the
+// data type encoded in the Key. The function returns the microservice
+// label, the data type and the list of parameters, that contains path
+// segments that follow the data path segment in the Key URL. The
+// parameter list is empty if data path is the Last segment in the
+// Key.
+//
+
+func ParseKey(key string) (label string, dataType string, name string) {
```

Comment:
This shouldn't be needed.. `models` package should be able to provide this with https://github.com/ligato/vpp-agent/blob/40bd6b1899e0abe9a30da72c5430225a44cf39b2/pkg/models/models.go#L100

which returns `registeredModel` that has these methods available (even though it's unexported..): https://github.com/ligato/vpp-agent/blob/40bd6b1899e0abe9a30da72c5430225a44cf39b2/pkg/models/spec.go#L68-L108

Or you can use the specific models directly: https://github.com/ligato/vpp-agent/blob/dev/api/models/vpp/interfaces/keys.go#L30

Reply:
Also these fields are available on the `registeredModel` too.. 

(note to self: that really has to be changed.. it's weird being used unexported...)

Reply:
Ok,
But I will look at this when I return from holiday.
I want to make a big refactor of this part when I return.